scripts/upgrade_admin_server.sh

```diff
@@ -20,6 +20,16 @@ set -x
 
 upgrade_admin_server()
 {
+    local installdir=/var/lib/crowbar/install
+
+    if [[ -f $installdir/admin_server_upgrading ]] ; then
+        echo "Exit: Upgrade already running..."
+        exit 1
```

Comment:
As this function is executed in background, we will  not know the exit value. So the check if the upgrade is already running should be done directly from crowbar, I think.